### PR TITLE
stable spirv-std 

### PR DESCRIPTION
**Pull request by VZout**
Allow compiling spirv-std on stable.

### CODE REVIEW

File: crates/spirv-std/src/storage_class.rs [hunk 1]
```diff
@@ -36,7 +36,7 @@ macro_rules! storage_class {
             /// Store the value in storage.
             #[inline]
             #[allow(unused_attributes)]
-            #[spirv(really_unsafe_ignore_bitcasts)]
+            #[cfg_attr(target_arch = "spirv", spirv(really_unsafe_ignore_bitcasts))]
```
Comment (eddyb) [inline crates/spirv-std/src/storage_class.rs:39]: Can you move the `allow` into the `cfg_attr`? I think a single `cfg_attr` supports multiple attributes nowadays, but the formatting might be a bit weird.

Comment (eddyb) [reply crates/spirv-std/src/storage_class.rs:39]: Also, this is a random place I ended up commenting, all of the changed attributes also have `allow`s.

Comment (VZout) [reply crates/spirv-std/src/storage_class.rs:39]: done

File: crates/spirv-std/src/lib.rs [hunk 2]
```diff
@@ -1,7 +1,10 @@
 #![no_std]
-#![feature(register_attr, repr_simd, core_intrinsics, lang_items)]
+#![cfg_attr(
+    target_arch = "spirv",
+    feature(register_attr, repr_simd, core_intrinsics, lang_items)
+)]
 #![cfg_attr(target_arch = "spirv", feature(asm))]
-#![register_attr(spirv)]
+#![cfg_attr(target_arch = "spirv", register_attr(spirv))]
```
Comment (eddyb) [inline crates/spirv-std/src/lib.rs:7]: I think this can be a single `cfg_attr` now.

File: crates/spirv-std/src/textures.rs [hunk 3]
```diff
@@ -1,22 +1,22 @@
 use glam::{Vec2, Vec3A, Vec4};
 
 #[allow(unused_attributes)]
-#[spirv(sampler)]
+#[cfg_attr(target_arch = "spirv", spirv(sampler))]
```
Comment (eddyb) [inline crates/spirv-std/src/textures.rs:4]: This file has a `allow`s everywhere as well.

File: crates/spirv-std/src/lib.rs (hunk 2 above)
Comment (VZout) [reply crates/spirv-std/src/lib.rs]: Missed this comment yesterday :)

**Review by eddyb — APPROVED** [review]

**Pull request merged**